token-client/src/test/java/com/sap/cloud/security/xsuaa/tokenflows/ClientCredentialsTokenFlowTest.java

```diff
@@ -89,20 +84,46 @@ public void execute_throwsIfServiceRaisesException() throws OAuth2ServiceExcepti
 
 	@Test
 	public void execute_withAdditionalAuthorities() throws TokenFlowException, OAuth2ServiceException {
-		OAuth2TokenResponse accessToken = new OAuth2TokenResponse(JWT_ACCESS_TOKEN, 441231, null);
-
-		Map<String, String> additionalAuthorities = new HashMap<String, String>();
+		OAuth2TokenResponse accessToken = mockRetrieveAccessToken();
+		Map<String, String> additionalAuthorities = new HashMap<>();
 		additionalAuthorities.put("DummyAttribute", "DummyAttributeValue");
 
-		Mockito.when(mockTokenService
+		OAuth2TokenResponse response = cut.attributes(additionalAuthorities).execute();
+
+		assertThat(response.getAccessToken(), is(accessToken.getAccessToken()));
+		verify(mockTokenService, times(1))
 				.retrieveAccessTokenViaClientCredentialsGrant(eq(TOKEN_ENDPOINT_URI), eq(clientCredentials),
-						isNull(), isNotNull()))
-				.thenReturn(accessToken);
+						isNull(), isNotNull(), anyBoolean());
+	}
+
+	@Test
+	public void execute_withCacheDisabled() throws TokenFlowException, OAuth2ServiceException {
+		OAuth2TokenResponse accessToken = mockRetrieveAccessToken();
+
+		OAuth2TokenResponse response = cut.disableCache(true).execute();
 
-		OAuth2TokenResponse jwt = cut.attributes(additionalAuthorities)
-				.execute();
+		assertThat(response.getAccessToken(), is(accessToken.getAccessToken()));
+		verifyTokenServiceCall(true);
 
-		assertThat(jwt.getAccessToken(), is(accessToken.getAccessToken()));
+		cut.disableCache(false).execute();
+
+		verifyTokenServiceCall(false);
+
+	}
+
+	private void verifyTokenServiceCall(boolean disableCache) throws OAuth2ServiceException {
```

Comment:
prefer this method name: verifyThatDisableCacheIs

Reply:
renamed! :)